nemo/collections/nlp/modules/common/megatron/adapters/parallel_adapters.py

```diff
@@ -301,15 +292,47 @@ def __init__(
             gradient_accumulation_fusion=gradient_accumulation_fusion,
         )
         # Setup adapter strategy
-        self.setup_adapter_strategy(adapter_strategy)
+        self.setup_adapter_strategy(adapter_mixin_strategies.ReturnResultAdapterStrategy())
+
+    def set_inference_table(self, prompt_representation: torch.Tensor):
+        """
+        This method caches the output representation from the Encoder and saves it inside `self.inference_table`.
+        """
+        prompt_representation = prompt_representation.detach().clone()
+        self.inference_table.set_prompt_table(prompt_representation)
+
+    def clear_inference_table(self,):
+        self.inference_table.clear_prompt_table()
+
+    def get_inference_table(self,):
+        return self.inference_table.get_prompt_table()
 
-    def forward(self, batch_size):
+    def _forward(self,):
```

Comment:
Better to not make this private for subclasses, rename to forward_inner()

Reply:
good point!